src/sentry/rules/processor.py

```diff
@@ -249,7 +257,56 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
                 else:
                     self.grouped_futures[key][1].append(rule_future)
 
-    def apply(self) -> Iterable[Any]:
+    def _get_active_release_rule_actions(self) -> Sequence[EventAction]:
+        # TODO: we need this to be configurable on a pre-project level?
+        return [
+            NotifyActiveReleaseEmailAction(
+                project=self.project, data={"targetType": ActionTargetType.RELEASE_MEMBERS.value}
+            )
+        ]
+
+    def apply_active_release_rule(
+        self,
+        conditions: Sequence[EventCondition],
+        filters: Sequence[EventFilter],
+        frequency_minutes: int = 5,
+    ) -> None:
+        now = timezone.now()
+        freq_offset = now - timedelta(minutes=frequency_minutes)
+        last_active_cache_key = "{}:p-{}:g-{}".format(
+            "active-release-last-active", self.event.project_id, self.event.group_id
+        )
+        last_active_time = cache.get(last_active_cache_key)
+
+        if last_active_time and last_active_time > freq_offset:
+            return
+
+        state = self.get_state()
+
+        if not all(
+            safe_execute(f.passes, event=self.event, state=state, _with_transaction=False)
+            for f in filters
+        ):
+            return
+
+        if not all(
+            safe_execute(c.passes, event=self.event, state=state, _with_transaction=False)
+            for c in conditions
+        ):
+            return
```

Comment:
One concern I have here is that this makes up to 4 queries whenever we run this rule. Since this will run on every event, it might be worth running this at most once per minute per group, regardless of whether we pass the filters and conditions. Given the nature of this feature, we don't really need to run this rule on every event for a group.

Reply:
Hmm, good suggestion. I included an extra throttling param to early-return on predicates evaluation.